integration/apps/rails-five/config/application.rb

```diff
@@ -0,0 +1,97 @@
+require_relative 'boot'
+
+# require "rails"
+# Pick the frameworks you want:
+require "active_model/railtie"
+require "active_job/railtie"
+require "active_record/railtie"
+require "action_controller/railtie"
+require "action_mailer/railtie"
+require "action_view/railtie"
+require "action_cable/engine"
+# require "sprockets/railtie"
+# require "rails/test_unit/railtie"
+
+# Require the gems listed in Gemfile, including any gems
+# you've limited to :test, :development, or :production.
+Bundler.require(*Rails.groups)
+
+class TraceMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    Datadog.tracer.trace('web.request', service: 'acme', resource: env['REQUEST_PATH']) do |span|
+      Datadog.runtime_metrics.associate_with_span(span)
+      @app.call(env)
+    end
+  end
+end
+
+class ShortCircuitMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    return [200, {}, []]
+  end
+end
+
+class ErrorMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    @app.call(env)
+    raise
+  end
+end
+
+class CustomError < StandardError
+  def message
+    'Custom error message!'
+  end
+end
+
+class CacheMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    request_id = env['action_dispatch.request_id']
+
+    # NOTE: Disabled for now, suspected to cause memory growth.
+    # Fetch from cache
+    # Rails.cache.fetch(request_id) { request_id }
```

Comment:
Interesting... Can you add a bit more context here?

Reply:
I think because each request ID is unique, and `fetch` gets or adds the request ID to the cache, it will grow indefinitely. This was added as a cheap way of implementing cache usage (for showing in traces) but it probably doesn't need to create this many keys.

Reply:
Your explanation makes some sense :)
May I suggest adding it as a comment? That would be a good starting point if someone ran into this and was also wondering what was the problematic effect.